lib/src/handlers/request_handler.dart

```diff
@@ -1,20 +1,43 @@
 import 'package:http_mock_adapter/http_mock_adapter.dart';
+import 'package:http_mock_adapter/src/Exceptions.dart';
+import 'package:http_mock_adapter/src/adapter_interface.dart';
+import 'package:http_mock_adapter/src/interceptors/http_interceptor.dart';
 
 /// The handler of requests sent by clients.
-class RequestHandler {
+class RequestHandler<T> {
   /// An HTTP status code such as - `200`, `404`, `500`, etc.
   int statusCode;
+  AdapterInterface _returnObject;
+  RequestHandler() {
+    /// If type parameter of the class is neither [DioAdapter] nor [MainInterceptor]
+    /// throws [RequestHandlerException]
+    if (T.runtimeType != DioAdapter || T.runtimeType != MainInterceptor)
+      RequestHandlerException();
+  }
 
   /// Map of <[statusCode], [data]>.
   final Map<int, dynamic> requestMap = {};
 
   /// Stores [response.data] in [requestMap] and returns [DioAdapter]
   /// the latter which is utilized for method chaining.
-  DioAdapter reply(int statusCode, dynamic data) {
+  AdapterInterface reply(int statusCode, dynamic data) {
     this.statusCode = statusCode;
 
     requestMap[this.statusCode] = data;
 
-    return DioAdapter();
+    // Checking the type of the `type parameter`
+    // and returning the relevant Class Instance
+    switch (T) {
```

Comment:
This does not work as intended.

If you remove the:

```dart
...
      default:
        _returnObject = DioAdapter();
...
```

or write `throw RequestHandlerException()` instead, the tests fail. By default, an exception should be thrown regardless.

Reply:
Why did you close this? This isn't resolved...